src/Optimizer/Configuration/AutoExtensionsConfiguration.php

```diff
@@ -0,0 +1,183 @@
+        ];
+    }
+
+    /**
+     * Validate an individual configuration entry.
+     *
+     * @param string $key   Key of the configuration entry to validate.
+     * @param mixed  $value Value of the configuration entry to validate.
+     * @return mixed Validated value.
+     */
+    protected function validate($key, $value)
+    {
+        switch ($key) {
+            case self::FORMAT:
+                if (! is_string($value)) {
+                    throw InvalidConfigurationValue::forInvalidSubValueType(
+                        self::class,
+                        self::FORMAT,
+                        'string',
+                        gettype($value)
+                    );
+                }
+
+                if (! in_array($value, Amp::FORMATS, true)) {
+                    throw InvalidConfigurationValue::forUnknownSubValue(
+                        self::class,
+                        self::FORMAT,
+                        Amp::FORMATS,
+                        $value
+                    );
+                }
+                break;
+
+            case self::AUTO_EXTENSION_IMPORT:
+                if (! is_bool($value)) {
+                    throw InvalidConfigurationValue::forInvalidSubValueType(
+                        self::class,
+                        self::AUTO_EXTENSION_IMPORT,
+                        'boolean',
+                        gettype($value)
+                    );
+                }
+                break;
+
+            case self::EXPERIMENT_BIND_ATTRIBUTE:
+                if (! is_bool($value)) {
+                    throw InvalidConfigurationValue::forInvalidSubValueType(
+                        self::class,
+                        self::EXPERIMENT_BIND_ATTRIBUTE,
+                        'boolean',
+                        gettype($value)
+                    );
+                }
+                break;
+
+            case self::EXTENSION_VERSIONS:
+                if (! is_array($value)) {
+                    throw InvalidConfigurationValue::forInvalidSubValueType(
+                        self::class,
+                        self::EXTENSION_VERSIONS,
+                        'array',
+                        gettype($value)
+                    );
+                }
+                break;
+
+            case self::IGNORED_EXTENSIONS:
+                if (! is_array($value)) {
+                    throw InvalidConfigurationValue::forInvalidSubValueType(
+                        self::class,
+                        self::IGNORED_EXTENSIONS,
+                        'array',
+                        gettype($value)
+                    );
+                }
+
+                // Assert that the extension names in the ignore list are valid extensions.
+                $reflection = new ReflectionClass(Extension::class);
+                $constants = $reflection->getConstants();
+
+                foreach ($value as $extension) {
+                    if (! in_array($extension, $constants)) {
```

Comment:
Array functions should always be used in strict mode.
```suggestion
                    if (! in_array($extension, $constants, true)) {
```